### PR TITLE
remove duplicate index urls

### PR DESCRIPTION
**Pull request by majuscule**
closes issue https://github.com/jazzband/pip-tools/issues/316

### CODE REVIEW

File: piptools/scripts/compile.py [hunk 1]
```diff
@@ -143,6 +143,8 @@ def cli(verbose, dry_run, pre, rebuild, find_links, index_url, extra_index_url,
         repository = LocalRequirementsRepository(existing_pins, repository)
 
     log.debug('Using indexes:')
+    # remove duplicate index urls before processing
+    repository.finder.index_urls = sorted(set(repository.finder.index_urls))
```
Comment (davidovich) [inline piptools/scripts/compile.py]: Thanks for your contribution!

As it is, this has the possibility of changing the index-url order, which is a property that needs to be maintained.

As a user, I might use index urls to explicitly set desired index usage ( A -> B -> C ).

See http://stackoverflow.com/a/39835527/28275 for a clean reference implementation that keeps order.

Comment (majuscule) [inline piptools/scripts/compile.py]: That's fine by me. I have pushed an update with the collections implementation as per your link.

**Comment by davidovich** [comment]
Thanks for this @majuscule

**Comment by majuscule** [comment]
Happy to help! If you have time, I would appreciate your thoughts on my comment to you here: https://github.com/jazzband/pip-tools/pull/418#issuecomment-285823330.

**Pull request merged**